cloud/ec2_client.go

```diff
@@ -513,6 +519,29 @@ func (c *awsClientImpl) DeleteKeyPair(ctx context.Context, input *ec2.DeleteKeyP
 	return output, nil
 }
 
+// GetProducts is a wrapper for pricing.GetProducts.
+func (c *awsClientImpl) GetProducts(ctx context.Context, input *pricing.GetProductsInput) (*pricing.GetProductsOutput, error) {
+	var output *pricing.GetProductsOutput
+	var err error
+	msg := makeAWSLogMessage("GetProducts", fmt.Sprintf("%T", c), input)
+	_, err = util.Retry(
+		func() (bool, error) {
+			output, err = c.pricing.GetProductsWithContext(ctx, input)
+			if err != nil {
+				if ec2err, ok := err.(awserr.Error); ok {
+					grip.Error(message.WrapError(ec2err, msg))
+				}
+				return true, err
+			}
+			grip.Info(msg)
```

Comment:
annotate more?

Reply:
This comes from `makeAWSLogMessage`, which is pretty annotated already.